### PR TITLE
[Merged by Bors] - feat(RingTheory/LocalProperties): being projective can be checked locally on stalks

### PR DESCRIPTION
**Pull request by erdOne**
... for finitely presented modules

Written at the AIM algebraic geometry workshop

Co-authored-by: David Swinarski

---

- [x] depends on: #18264

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/FinitePresentation.lean
Comment (chrisflav) [inline Mathlib/Algebra/Module/FinitePresentation.lean:325]: This declaration shares most of its variables with the one above, can you make them into one section?

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:13]:
```suggestion
# Local properties of projective modules

```
copy paste leftover?

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:27]: Should this be in the `Module` namespace?

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:20]: This is listed twice, did you mean to include something else here?

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:108]: Can this go to an earlier file?

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:4]: David Swinarski is listed as Co-Author of the PR, maybe then also add them here?

File: Mathlib/RingTheory/LocalProperties/Submodule.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Submodule.lean:10]:
```suggestion
# Local properties of submodules
```

File: Mathlib/RingTheory/LocalProperties/Submodule.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Submodule.lean:16]:
```suggestion
```
The `section` does not seem to do anything

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:86]: Maybe also a general `IsLocalizedModule` version is useful?

File: Mathlib/RingTheory/LocalProperties/Submodule.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Submodule.lean:36]: The docstrings in this file seem to all be about the version for `Ideal`s. For the sake of intuitive notation, could you also replace `I` and `J` (which allude to ideals) by e.g. `N` and `O`?

**Comment by erdOne** [comment]
Thanks for the review!

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:18]:
```suggestion
- `LinearMap.split_surjective_of_localization_maximal` If `N` is finitely presented, then `f : M →ₗ[R] N`
  being split injective can be checked on stalks (of maximal ideals).
```

File: Mathlib/RingTheory/LocalProperties/Submodule.lean
Comment (chrisflav) [inline Mathlib/RingTheory/LocalProperties/Submodule.lean:63]:
```suggestion
/-- A submodule is trivial if its localization at every maximal ideal is trivial. -/
```

**Review by chrisflav — APPROVED** [review]

**Comment by chrisflav** [comment]
The long line linter complains, otherwise I am happy.

**Comment by chrisflav** [comment]
Thanks!

maintainer merge

**Comment by erdOne** [comment]
Okay I think I have a better approach. Let's hold this for now.

**Comment by erdOne** [comment]
No I was wrong :((
Let's keep this open.

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (jcommelin) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:125]: At some point we might want to have some syntax for variable blocks like this...

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: Mathlib/RingTheory/LocalProperties/Projective.lean
Comment (Vierkantor) [inline Mathlib/RingTheory/LocalProperties/Projective.lean:39]: The unused variable linter flagged this.
```suggestion
    (H : ∀ (I : Ideal R), I.IsMaximal →
```

**Review by Vierkantor — APPROVED** [review]
Could you please merge the `master` branch into this PR and then (assuming everything builds) reply with `bors r+`?

bors d+

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**
